src/dataset.rs

```diff
@@ -1,4 +1,9 @@
-use std::{ffi::CString, ffi::NulError, path::Path, ptr};
+use std::{
+    ffi::NulError,
+    ffi::{CStr, CString},
```

Comment:
Nit: let's merge these:

```suggestion
    ffi::{CStr, CString, NulError},
```

and add a changelog entry.